### PR TITLE
Add acl switch [consul] (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by zhenik**
Just to verify the idea first

### CODE REVIEW

File: ansible/templates/consul.service.j2 [hunk 1]
```diff
@@ -8,7 +8,7 @@ After=network-online.target
 EnvironmentFile=-/etc/sysconfig/consul
 Environment=GOMAXPROCS=2
 Restart=on-failure
-ExecStart=/usr/local/bin/consul agent -dev -config-file=/etc/consul.d/config.hcl $OPTIONS
+ExecStart=/usr/local/bin/consul agent -dev -config-file=/etc/consul.d/ $OPTIONS
```
Comment (fredrikhgrelland) [inline ansible/templates/consul.service.j2:11]:
```suggestion
ExecStart=/usr/local/bin/consul agent -dev -config-dir=/etc/consul.d/ $OPTIONS
```

File: ansible/playbook.yml [hunk 2]
```diff
@@ -18,6 +18,10 @@
         src: startup.yml
         dest: /etc/ansible/startup.yml
 
+    # ACL
+    - name: Copy acl template config
+      template: src=acl-consul.hcl.j2 dest=/etc/ansible/config-merge1.hcl
+
```
Comment (fredrikhgrelland) [inline ansible/playbook.yml:24]: This should be the actual template?

Comment (fredrikhgrelland) [reply ansible/playbook.yml:24]: Not sure if this makes sense

File: ansible/startup.yml [hunk 3]
```diff
@@ -7,6 +7,17 @@
...
   tasks:
+    # create additional config with acl=deny for configuration merge
+    # https://www.consul.io/docs/agent/options.html
+    - set_fact: consul_acl="allow"
+      when: consul_acl is undefined
+    - debug: msg="ACL default policy {{ consul_acl }}"
+    - name: "Verify ACL default strategy"
+      copy:
+        src: /etc/ansible/config-merge1.hcl
+        dest: /etc/consul.d/config-merge1.hcl
```
Comment (fredrikhgrelland) [inline ansible/startup.yml:18]: I would use templating here, rather than in playbook. would be self-contained.

**Comment by zhenik** [comment]
Does the current approach in this PR is what do you expect @fredrikhgrelland or you have another idea how to achieve `ACL switch`?

**Comment by fredrikhgrelland** [comment]
> Does the current approach in this PR is what do you expect @fredrikhgrelland or you have another idea how to achieve `ACL switch`?

Did you see comments in review? In short, Yes. But I think it could be simplified. Only one template building it env switch in startup.yml

**Comment by m88614** [comment]
Are you doing this only for consul in this first PR?

**Comment by zhenik** [comment]
> Are you doing this only for consul in this first PR? Or are you planning to extend it to nomad as well?

Consul only, for now. But I want to resolve #75  before this PR.

**Comment by m88614** [comment]
> > Are you doing this only for consul in this first PR? Or are you planning to extend it to nomad as well?
> 
> Consul only, for now. But I want to resolve #75 before this PR.

Got it. I removed the issue I linked then.

**Pull request closed without merging**